cyral/resource_cyral_sidecar_listener.go

```diff
@@ -0,0 +1,422 @@
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string),
+						d.Get(ListenerIdKey).(string))
+
+				},
+				NewResourceData: func() ResourceData { return &SidecarListenerResource{} },
+			}, ReadSidecarListenersConfig,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "SidecarListenersResourceDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string),
+						d.Get(ListenerIdKey).(string))
+				},
+			},
+		),
+		Schema: map[string]*schema.Schema{
+			ListenerIdKey: {
+				Description: "ID of the listener that will be bound to the sidecar.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			SidecarIdKey: {
+				Description: "ID of the sidecar that the listener will be bound to.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			RepoTypesKey: {
+				Description: "List of repository types that the listener supports. Currently limited to one repo type, eg [\"mysql\"].",
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			UnixListenerKey: {
+				Description: "Unix listener settings.",
+				Type:        schema.TypeSet,
+				Optional:    true,
+				// Notice the MaxItems: 1 here. This ensures that the user can only specify one this block.
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						FileKey: {
+							Description: "File in which the sidecar will listen for the given repository.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+					},
+				},
+			},
+			MultiplexedKey: {
+				Description: "Multiplexed listener, defaults to not multiplexing (false). Not supported for all repository types.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			TcpListenerKey: {
+				Description: "TCP listener settings.",
+				Type:        schema.TypeSet,
+				Optional:    true,
+				// Notice the MaxItems: 1 here. This ensures that the user can only specify one this block.
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						HostKey: {
+							Description: "Host in which the sidecar will listen for the given repository. Omit to listen on all interfaces.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						PortKey: {
+							Description: "Port in which the sidecar will listen for the given repository.",
+							Type:        schema.TypeInt,
+							Required:    true,
+						},
+					},
+				},
+			},
+			MysqlSettingsKey: {
+				Description: "MysqlSettings represents the listener settings for a [mysql, galera, mariadb] data repository.",
```

Comment:
This is typically spelled as `MySQL`. Since this is a user-facing doc, we should probably stick to that.

Same for the names in the square brackets (they are proper nouns, so we should keep the names as they are typically spelled).

Reply:
The values in brackets here refer to the repo type, which is all-lower. Not sure if we want to mix those? 
Anyways, for the rest, you are suggesting 
`MysqlSettingsKey    = "mysql_settings"` -> `MysqlSettingsKey    = "MySQL_settings" `
and the decription to reflect that, ie `"MySQL_settings represents ..."`
or am I misreading it?

Reply:
updated according to comments from Victor, marking as resolved.